UI: Add "More Info" Button to OBS Basic Stream Settings

mv2731 opened the pull request:
Add "More Info" Button to OBS Basic Stream Settings by doing the following:

- Add UpdateMoreInfoLink function to window-basic-settings-stream.cpp
- Add slots and signals to trigger calling of UpdateMoreInfoLink 
when a service is selected from the dropdown
- Include UpdateMoreInfoLink function in the window-basic-settings.hpp file
to make it available to the class
- Add the button itself to the stream settings UI and set its size policy 
to maximum to ensure that it doesn't expand when the window does
- Test manually to make sure everything works as expected
- Ran clang-format on all c and cpp files

mv2731 committed:
UI: Add UpdateMoreInfoLink() to basic

Add UpdateMoreInfoLink() function to basic settings
from the auto-config-stream page. Call this function
whenever "More Info" button is pressed using slots
and signals.

mv2731 committed:
UI: Add "More Info" to basic settings UI

Add "More Info" button to basic settings UI
by adding a combo Widget including both the
service dropdown and the "More Info" button
in order to keep the layout as a "QFormLayout."

mv2731 committed:
UI: Apply clang-format

mv2731 committed:
UI: Set "More Info" size policy to maximum

The "More Info" button used to expand if the
settings window size changed, to occupy as much
space as the service dropdown at any time, which
looks silly. Set the "More Info" size policy to
maximum to make it stop expanding when settings
window size is changed.

mv2731 committed:
UI: Check diffs before making PR

ushadow commented on UI/window-basic-settings-stream.cpp:
```diff
@@ -229,6 +232,35 @@ void OBSBasicSettings::SaveStream1Settings()
+	}
+
+	QString serviceName = ui->service->currentText();
+	obs_properties_t *props = obs_get_service_properties("rtmp_common");
+	obs_property_t *services = obs_properties_get(props, "service");
```
These two lines are not needed.

ushadow commented:
It looks great! Could update the link to "https://developers.google.com/youtube/v3/live/guides/ingestion-protocol-comparison"? Thanks!

mv2731 committed:
rtmp-services: Replace more_info_link

Replace the old more_info_link for testing
with the actual link leading to documentation
about HLS as an ingestion protocol.
